src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -173,7 +181,7 @@ private TransactionV0 toXdr() {
     transaction.setSourceAccountEd25519(StrKey.encodeToXDRAccountId(this.mSourceAccount).getAccountID().getEd25519());
     transaction.setOperations(operations);
     transaction.setMemo(mMemo.toXdr());
-    transaction.setTimeBounds(mTimeBounds == null ? null : mTimeBounds.toXdr());
+    transaction.setTimeBounds(preconditions == null || preconditions.getTimeBounds() == null ? null : preconditions.getTimeBounds());
```

Comment:
Preconditions should never be null, as timebounds are required, so then we can unconditionally do `preconditions.getTimebounds()`, I think.

Reply:
I've got builder/conditions/tx aligned mostly to this rule with latest commits, but, I ran into a couple existing tests that demonstrated building xdr tx's with no timebounds based on the notion of 'timeout' being set from builder as an alternative to timebounds, such as [testParseV0Transaction on master branch](https://github.com/stellar/java-stellar-sdk/blob/master/src/test/java/org/stellar/sdk/TransactionTest.java#L336), so, I've retained the ability to construct a tx with no/null timebounds to support those test cases which asserted on the serialized/base64 xdr.